### DISCUSSION TITLE
Implement import from MAPS.ME

### DISCUSSION THREAD

**Discussion by rtsisyk** [organicmaps/organicmaps#380] [Ideas]
Add a button to import all bookmarks from MAPS.ME.

**Comment by rtsisyk** [comment]
https://github.com/organicmaps/organicmaps/pull/416

**Reply by GhostInTheKayak** [reply]
Only Android?

**Reply by biodranik** [reply]
It's not technically possible on iOS, sorry.

**Reply by seagullua** [reply]
@biodranik It can be implemented in ios.
On ios the app can have a link onto the website where you can upload the file with bookmarks. The website will give a code in text format (for example json). You will have to copy and paste the code into the input inside the app.

**Reply by GhostInTheKayak** [reply]
This is the killer. I have Maps.Me on an iPad with ~120 groups containing ~4000 bookmarks, so I really need some way to export them all and import them into OM.
Any other thoughts?

**Reply by biodranik** [reply]
Do you see your bookmarks there in the File Sharing on the desktop?

**Reply by GhostInTheKayak** [reply]
@seagullua -- which website?

@biodranik -- iTunes does allow me to export the KMB files from the iPad to my PC, but the KMB files are of no use unless I can either
- convert these KMB files into KMZ files
- import the KMB file directly into OM

BTW: Can I the export the bookmark groups in OM as KMZ files?

**Reply by biodranik** [reply]
KMB to KML is possible in different ways:
1. Export each list manually
2. Put kmb files in Android device and press Import button in Bookmarks in OM for Android.
3. Build a patched version of our code which will load also KMB in addition to KML (see [here](https://github.com/organicmaps/organicmaps/blob/b97b57480c576ceaac46468ff453b4da389a6fd1/map/bookmark_manager.cpp#L1913))

OM exports KMZ, of course.

**Reply by GhostInTheKayak** [reply]
Any thoughts on importing the KML files into OM on an iPad?

**Reply by biodranik** [reply]
What is wrong with importing KML on an iPad? It should work.

**Reply by GhostInTheKayak** [reply]
I agree, it should be possible, but **how** ? There is no import option that I can find.

**Comment by biodranik** [comment]
There is no button in OM "import from mapsme". Open mapsme, export bookmarks from there, and immediately open them in OM.

**Reply by GhostInTheKayak** [reply]
Wow! Thanks.
Sort-of obvious really, but only when you have tried it.
So, I will set aside an afternoon to do that for my 200+ groups, but at least it is a once-only thing!

**Reply by Jeremboo** [reply]
I was looking for that feature but this solution worked perfectly! Thanks!

**Comment by biodranik** [comment]
Another way is to copy bookmark folder directly from iPad using Finder. Select File Sharing and select MapsMe.

**Reply by GhostInTheKayak** [reply]
Thanks, but sadly, Finder is not available on Windows. :-(
... or is it??

**Reply by biodranik** [reply]
No, iOS File Sharing with apps works only on Mac.

**Comment by RositaMagelinskaite** [comment]
HELP! Lost all of the bookmarks after actualization of my Android system. Found them in my archives but they are in KMB format. It seems that I could do something, meaning get them back, if I convert them to KML or KMZ format but I cannot find a way of how to do it. Could anyone guide me, please??

**Reply by biodranik** [reply]
You can build and run kmb_to_kml converter in our repository, or you can send me your bookmarks in KMB, and I can help to convert them: `alex at organicmaps dot app`

**Reply by emiliozgz** [reply]
Hi @biodranik, I am struggling to recover 3 .kmb files in the new Maps.me (before I just overwrote it). Do you say that you could convert them to .kml? In that case let me know how can I send them to you, I would really appreciate it! :-)

**Reply by biodranik** [reply]
My email is written above )

**Reply by midorimelanie** [reply]
Hello ! I'm also in the same situation : some time back, after an update, all my bookmarks on maps me were converted to kmb files and I just don't manage converting them back to kml or gpx files. I've tried importing them on OM but it's not working. Would you mind helping me out too ? Thank you !

**Reply by biodranik** [reply]
@midorimelanie send me kmb files by email above.

**Comment by dantalian-pv** [comment]
I have found one workaround. If you open bookmarks in MapsMe, then open any list, there will be "Share" button. If you clink on it there will be Organic Maps in the list of the apps into which you can "Share" the bookmarks. By clicking on it all my bookmarks together with lists were successfully copied to Organic Maps.

**Comment by RositaMagelinskaite** [comment]
Hey guys. A year and a few months have gone by and I am having problems with maps.me app again. Yesterday everything was fine, all bookmarks were in place. Today - basically everything's gone, only 3 folders remain and their names have a bunch of "111111" in the end. For instance, "Belgium111111111". WHAT HAPPENED? No updates happened, my phone is with a lock screen, so it could have not been "pocket's job".  Worst thing is that I think those bookmarks of the last year and a few months have not been saved as any file on my phone. Any insight? Any ideas for what I could do?

**Reply by McLovin654** [reply]
Did you found a solution for this? Exactly same happened to me today... Most of the bookmarks are gone and folders have these 1111 in the end. I'm currently traveling in South America and have lost all my bookmarked places where I've been with all the notes and everything... Ouch

**Reply by biodranik** [reply]
Try to import bookmarks using "Import" button in Bookmark and tracks dialog in Organic Maps Beta https://cdn.organicmaps.app/apk/2023.10.15-20-Google-beta.apk

The next release will have a fixed KMB import.

You can also try to find kmb files and send them to us, maybe we can help to repair them.

**Reply by McLovin654** [reply]
Well I have some KMB files but only a few which are not so important to me. Most of the important bookmarks just disappeared. Some guy here was talking about that he found them "in the archives" but I don't know what he means by that. I searched maps.me folder and they are not there. Is there any other place I could find it?

**Reply by RositaMagelinskaite** [reply]
Hey McLovin, no, unfortunately, I did not find a solution for this. Did you manage to move forward with this?

**Comment by biodranik** [comment]
We have fixed the non-working import, it should be working again in the upcoming release. Manual conversion of KMB to KML should also work using kmb_to_kml tool from the repository.

**Reply by xhanacek** [reply]
Dear Alexander, it's too complicated for me. Could I send you KMB to convert to KML?

**Reply by biodranik** [reply]
Yes, of course.

**Comment by McLovin654** [comment]
Hi, well now I'm trying to export my bookmarks like everyday so when
something similar happens again I'll have some backup... But the app was
working fine since that mysterious issue so I'm good now.

Do you plan maybe automatic backup in the future?

Roman

**Reply by biodranik** [reply]
Do you remember what exactly happened? What exactly did you do when it happened?

Backup is planned, see #2082
